pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -345,31 +255,47 @@ func (r *repository) Put(manifest *schema1.SignedManifest) error {
 				},
 			},
 			DockerImageReference: fmt.Sprintf("%s/%s/%s@%s", r.registryAddr, r.namespace, r.name, dgst.String()),
-			DockerImageManifest:  string(manifest.Raw),
+			DockerImageManifest:  string(payload),
 		},
 	}
 
-	if err := r.fillImageWithMetadata(manifest, &ism.Image); err != nil {
-		return err
+	for _, option := range options {
+		if opt, ok := option.(distribution.WithTagOption); ok {
```

Comment:
I'm really afraid that in some future versions of docker, the daemon will:

1. push by digest using `PUT /v2/<name>/manifests/<digest>`
2. tag the pushed manifest using some other endpoint

We should handle this case as well. Defaulting to `latest` is not correct here. We should just create a new image object without tagging it into any image stream if the tag is missing.